ConfigSpace/hyperparameters.pyx

```diff
@@ -877,12 +888,283 @@ cdef class NormalFloatHyperparameter(FloatHyperparameter):
+            if self.q is not None:
+                self._lower = self.lower - (self.q / 2. - 0.0001)
+                self._upper = self.upper + (self.q / 2. - 0.0001)
+            else:
+                self._lower = self.lower
+                self._upper = self.upper
+        
+        self.default_value = self.check_default(default_value)
+        self.normalized_default_value = self._inverse_transform(self.default_value)
+
+        if self.q is not None:
+            # There can be weird rounding errors, so we compare the result against self.q, see
+            # In [13]: 2.4 % 0.2
+            # Out[13]: 0.1999999999999998
+            if np.round((self.upper - self.lower) % self.q, 10) not in (0, self.q):
+                raise ValueError(
+                    'Upper bound (%f) - lower bound (%f) must be a multiple of q (%f)'
+                    % (self.upper, self.lower, self.q)
+                )
+
+    def __repr__(self) -> str:
+        repr_str = io.StringIO()
+        repr_str.write("%s, Type: BetaFloat, Alpha: %s Beta: %s, Range: [%s, %s], Default: %s" % (self.name, repr(self.alpha), repr(self.beta), repr(self.lower), repr(self.upper), repr(self.default_value)))
+
+        if self.log:
+            repr_str.write(", on log-scale")
+        if self.q is not None:
+            repr_str.write(", Q: %s" % str(self.q))
+        repr_str.seek(0)
+        return repr_str.getvalue()
+
+    def __eq__(self, other: Any) -> bool:
+        """
+        This method implements a comparison between self and another
+        object.
+
+        Additionally, it defines the __ne__() as stated in the
+        documentation from python:
+            By default, object implements __eq__() by using is, returning NotImplemented
+            in the case of a false comparison: True if x is y else NotImplemented.
+            For __ne__(), by default it delegates to __eq__() and inverts the result
+            unless it is NotImplemented.
+
+        """
+        if not isinstance(other, self.__class__):
+            return False
+
+        return (
+            self.name == other.name and
+            self.default_value == other.default_value and
+            self.alpha == other.alpha and
+            self.beta == other.beta and
+            self.log == other.log and
+            self.q == other.q and
+            self.lower == other.lower and
+            self.upper == other.upper
+        )
+
+    def __copy__(self):
+        return BetaFloatHyperparameter(
+            name=self.name,
+            default_value=self.default_value,
+            alpha=self.alpha,
+            beta=self.beta,
+            log=self.log,
+            q=self.q,
+            lower=self.lower,
+            upper=self.upper,
+            meta=self.meta
+        )
+
+    def __hash__(self):
+        return hash((self.name, self.alpha, self.beta, self.lower, self.upper, self.log, self.q))
+
+    def to_uniform(self) -> 'UniformFloatHyperparameter':
+        lb = self.lower
```

Comment:
Is there a reason you don't pass these directly into the constructor?

Reply:
No. This was just following what seemed like convention (see the NormalFloat and NormalInt)

Reply:
I see. If possible, could you please change that.